cmd/katapult/data_centers_test.go

```diff
@@ -0,0 +1,157 @@
+		Name:      "hello",
+		Permalink: "POG1",
+		Country: &core.Country{
+			ID:   "POG",
+			Name: "Pogland",
+		},
+	},
+	{
+		ID:        "dc_9UVoPiUQoI1cqtR0",
+		Name:      "hello",
+		Permalink: "GB1",
+		Country: &core.Country{
+			ID:   "UK",
+			Name: "United Kingdom",
+		},
+	},
+}
+
+type mockDataCentersClient struct {
+	dcs    []*core.DataCenter
+	throws string
+}
+
+func (m mockDataCentersClient) List(context.Context) ([]*core.DataCenter, *katapult.Response, error) {
+	if m.throws != "" {
+		return nil, nil, errors.New(m.throws)
+	}
+	return m.dcs, nil, nil
+}
+
+func (m mockDataCentersClient) Get(
+	_ context.Context, ref core.DataCenterRef) (*core.DataCenter, *katapult.Response, error) {
+	if m.throws != "" {
+		return nil, nil, errors.New(m.throws)
+	}
+	for _, v := range m.dcs {
+		if v.Permalink == ref.Permalink {
+			return v, nil, nil
+		}
+	}
+	return nil, nil, fmt.Errorf("unknown datacentre")
+}
+
+func TestDataCenters_List(t *testing.T) {
+	tests := []struct {
+		name string
+
+		dcs     []*core.DataCenter
+		want    string
+		stderr  string
+		throws  string
+		wantErr string
+	}{
+		{
+			name: "data center list",
+			dcs:  fixtureDataCenters,
+			want: ` - hello (POG1) [dc_9UVoPiUQoI1cqtRd] / Pogland
+ - hello (GB1) [dc_9UVoPiUQoI1cqtR0] / United Kingdom
+`,
+		},
+		{
+			name: "empty data centers",
+			dcs:  []*core.DataCenter{},
+		},
+		{
+			name:    "data center error",
+			throws:  "test error",
+			wantErr: "test error",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cmd := dataCentersCmd(mockDataCentersClient{dcs: tt.dcs, throws: tt.throws})
+			stdout := &bytes.Buffer{}
+			stderr := &bytes.Buffer{}
+			cmd.SetOut(stdout)
+			cmd.SetErr(stderr)
+			cmd.SetArgs([]string{"list"})
+			err := cmd.Execute()
+
+			if tt.wantErr != "" {
+				require.EqualError(t, err, tt.wantErr)
+				return
+			}
+			require.NoError(t, err)
+
+			assert.Equal(t, tt.want, stdout.String())
+			assert.Equal(t, tt.stderr, stderr.String())
+		})
+	}
+}
+
+func TestDataCenters_Get(t *testing.T) {
+	tests := []struct {
+		name string
+
+		args    []string
+		dc      string
+		want    string
+		stderr  string
+		wantErr string
+	}{
+		{
+			name: "display POG1",
+			args: []string{"get", "POG1"},
+			want: "hello (POG1) [POG1-ID] / Pogland\n",
+		},
+		{
+			name: "display GB1",
+			args: []string{"get", "GB1"},
+			want: "hello (GB1) [GB1-ID] / United Kingdom\n",
+		},
+		{
+			name:    "display invalid DC",
+			args:    []string{"get", "UNPOG1"},
+			stderr:  "Error: unknown datacentre\n",
+			wantErr: "unknown datacentre",
+		},
+	}
+
+	cmd := dataCentersCmd(mockDataCentersClient{dcs: fixtureDataCenters})
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			stdout := &bytes.Buffer{}
+			stderr := &bytes.Buffer{}
+			cmd.SetOut(stdout)
+			cmd.SetErr(stderr)
+			cmd.SetArgs(tt.args)
+			err := cmd.Execute()
+			assert.Equal(t, tt.stderr, stderr.String())
```

Comment:
Unless I'm blind, I think `tt.want` is never asserted against `stdout` in this test. Also I think we could use the `testAssertCommand()` helper here like in networks and orgs tests :)